### ISSUE TITLE
Object Detection: Precision 1 Recall 0

### ISSUE THREAD

**Issue by fninsiima** [ultralytics/ultralytics#18795]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I am training an object detector however the results show a precision of 1 and a recall of 0 for some classes.
I am not certain of whether this is a bug or not.
It would be helpful to know what the cause of this issue is and/or how to migitate it.

<img width="811" alt="Image" src="https://github.com/user-attachments/assets/736bf807-8e37-47d6-b415-8459263769a5" />

### Environment

```
Ultralytics 8.3.65 🚀 Python-3.11.11 torch-2.5.1+cu121 CUDA:0 (NVIDIA A100-SXM4-40GB, 40514MiB)
Setup complete ✅ (12 CPUs, 83.5 GB RAM, 87.2/235.7 GB disk)

OS                  Linux-6.1.85+-x86_64-with-glibc2.35
Environment         Colab
Python              3.11.11
Install             pip
RAM                 83.48 GB
Disk                87.2/235.7 GB
CPU                 Intel Xeon 2.20GHz
CPU count           12
GPU                 NVIDIA A100-SXM4-40GB, 40514MiB
GPU count           1
CUDA                12.1
```

### Minimal Reproducible Example

```
# import YOLO model
from ultralytics import YOLO

# Load a model
model = YOLO('yolo11n.pt') # load a pretrained model (recommended for training)

# Train the model
model.train(data='/content/data.yaml', epochs=20, imgsz=800,
            classes=[2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,
                     38,39,43,44,46,48])
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fninsiima, thank you for raising this issue and contributing to Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) to explore helpful resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These might address common questions you may have.

From your description, this may be a 🐛 **Bug Report**. If so, your inclusion of the Minimal Reproducible Example (MRE) is fantastic – thank you! However, if there are additional details related to data preparation, ground truth labels, or evaluation metrics setup that might be relevant, please do share those as well so we can better assist. A detailed MRE helps streamline debugging! For more guidance, check out our [minimum reproducible example guide](https://docs.ultralytics.com/help/minimum-reproducible-example/).

If this is a custom training ❓ **Question**, the issue you’re observing (precision of 1 and recall of 0 for some classes) could result from your dataset or model configuration. Please share more details about your dataset (e.g., class distribution, any annotation samples, etc.) and any related training logs. Also, make sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimal settings and insights.

### Next Steps

We recommend upgrading to the latest version of Ultralytics to ensure all fixes and features are applied. To upgrade:

```bash
pip install -U ultralytics
```

YOLO can be run in the following 🖥️ environments, which are pre-configured to help eliminate setup-related issues. Let us know if the problem persists in any of these:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Refer to [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: Refer to [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Refer to [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community & Support

Join the Ultralytics community 🎉 and share your insights or queries:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for discussion forums
- [Subreddit](https://reddit.com/r/Ultralytics) for connecting with other users 📰

Rest assured, our team 👩‍💻👨‍💻 will review your report and respond shortly. Thank you for being part of the Ultralytics community!

**Comment by Y-T-G** [comment]
It's not a bug. It's based on the performance of the model. The number of epochs you trained is very low.

**Comment by fninsiima** [comment]
> It's not a bug. It's based on the performance of the model. The number of epochs you trained is very low.

The reason as to why I suspected it might be a bug is because I first did the training with yolov8.
It still gave me precision 1 and recall 0 but only on a few classes i.e. 2
Then when I tried out yolov11, the number of instances of precision 1 and recall 0 increased.

Attached is the performance of the yolov8 model.

<img width="843" alt="Image" src="https://github.com/user-attachments/assets/38915a2b-457b-4b80-a920-faf327686a04" />

**Comment by Y-T-G** [comment]
It doesn't necessarily mean it's a bug. Different models learn differently.
